scripts/lib-franklin.js

```diff
@@ -775,6 +775,30 @@ export function decorateRenderHints(block) {
 export function isInternalPage() {
   return getHref().indexOf('/sidekick/blocks/') > 0 || getHref().indexOf('/_tools/') > 0;
 }
+
+function sendResizeEvent() {
+  const heightPx = window.innerHeight || document.documentElement.clientHeight;
+  const widthPx = window.innerWidth || document.documentElement.clientWidth;
+
+  const baseFontSize = parseFloat(getComputedStyle(document.documentElement).fontSize);
+  const heightRem = heightPx / baseFontSize;
+  const widthRem = widthPx / baseFontSize;
+
+  // Create a new custom event with custom parameters
+  const resizeEvent = new CustomEvent('viewportResize', {
+    detail: {
+      heightPx,
+      widthPx,
+      heightRem,
+      widthRem,
+    },
+  });
+
+  window.dispatchEvent(resizeEvent);
+}
+
+window.addEventListener('resize', sendResizeEvent);
```

Comment:
but if we go with matchMedia event, we have to check on page load. Or set set up an interval if want to check periodically. Or I am missing something.

Also, window.resize is an stanard event that has been used widely across projects https://github.com/search?q=org%3Ahlxsites+window.addEventListener%28%27resize%27&type=code

Let me know if we should still go with window matchMedia

Reply:
I think `matchMedia` also has a [change event](https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList/change_event). For e.g.

```
const widerScreenWidth = window.matchMedia('(min-width: 77rem)');
  widerScreenWidth.addEventListener('change', (event) => {
```

> window.resize is an stanard event that has been used widely across projects

Yes I see both `matchMedia` and `resize` being used commonly across `hlxsites`. I had an impression that `resize` may be costlier. [Related article](https://tigeroakes.com/posts/resize-event-alternatives/#:~:text=The%20matchMedia%20API%20lets%20you,when%20the%20breakpoint%20is%20passed.)  But maybe I am just splitting hairs since I haven't quantified it. (@hparwani2  do you recollect ?) 

Since were already using `matchMedia` in header.js, maybe we can reuse. Up to you, whichever you deem fit.